x-pack/plugin/deprecation/src/main/java/org/elasticsearch/xpack/deprecation/DeprecationInfoAction.java

```diff
@@ -230,6 +236,25 @@ public int hashCode() {
         }
     }
 
+    private static ClusterState removeSkippedSettings(ClusterState state, List<String> skipTheseDeprecatedSettings) {
```

Comment:
Since it was private, I tested it through the from() method. That's actually really all that testFromWithHiddenSettings() is testing. Maybe I should just rename that testRemoveSkippedSettings() and kill two birds with one stone.

Reply:
OK, renamed it to make it more clear that that's what I'm testing.